launcher/ui/pages/instance/InstanceSettingsPage.cpp

```diff
@@ -437,6 +454,70 @@ void InstanceSettingsPage::on_javaTestBtn_clicked()
     checker->run();
 }
 
+void InstanceSettingsPage::updateAccountsMenu()
+{
+    accountMenu->clear();
+
+    auto accounts = APPLICATION->accounts();
+    int accountIndex = accounts->findAccountByProfileId(m_settings->get("InstanceAccountId").toString());
+
+    if (accountIndex != -1)
+    {
+        auto account = accounts->at(accountIndex);
+        ui->instanceAccountSelector->setText(account->profileName());
+        ui->instanceAccountSelector->setIcon(account->getFace());
+    } else {
+        ui->instanceAccountSelector->setText(tr("No default account"));
+        ui->instanceAccountSelector->setIcon(APPLICATION->getThemedIcon("noaccount"));
+    }
+
+    for (int i = 0; i < accounts->count(); i++)
+    {
+        MinecraftAccountPtr account = accounts->at(i);
+        QAction *action = new QAction(account->profileName(), this);
+        action->setData(i);
+        action->setCheckable(true);
+        if (accountIndex == i)
+        {
+            action->setChecked(true);
+        }
+
+        auto face = account->getFace();
+        if(!face.isNull()) {
+            action->setIcon(face);
+        }
+        else {
+            action->setIcon(APPLICATION->getThemedIcon("noaccount"));
+        }
+
+        accountMenu->addAction(action);
+        connect(action, SIGNAL(triggered(bool)), SLOT(changeInstanceAccount()));
+    }
+}
+
+void InstanceSettingsPage::changeInstanceAccount()
+{
+    QAction *sAction = (QAction *)sender();
+
+    // Profile's associated Mojang username
+    if (sAction->data().type() != QVariant::Type::Int)
+        return;
```

Comment:
```suggestion
    Q_ASSERT(sAction->data().type() == QVariant::Type::Int);
```
seems to me that this should never not be an integer